osquery/tables/forensic/sleuthkit.cpp

```diff
@@ -0,0 +1,347 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <map>
+#include <set>
+
+#include <boost/noncopyable.hpp>
+#include <boost/filesystem.hpp>
+
+#include <tsk/libtsk.h>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+const std::map<TSK_FS_META_TYPE_ENUM, std::string> kTSKTypeNames{
+    {TSK_FS_META_TYPE_REG, "regular"},
+    {TSK_FS_META_TYPE_DIR, "directory"},
+    {TSK_FS_META_TYPE_LNK, "symlink"},
+    {TSK_FS_META_TYPE_BLK, "block"},
+    {TSK_FS_META_TYPE_CHR, "character"},
+    {TSK_FS_META_TYPE_FIFO, "fifo"},
+    {TSK_FS_META_TYPE_SOCK, "socket"},
+};
+
+class DeviceHelper : private boost::noncopyable {
+ public:
+  DeviceHelper(const std::string& device_path)
+      : image_(std::make_shared<TskImgInfo>()),
+        volume_(std::make_shared<TskVsInfo>()),
+        device_path_(device_path) {}
+
+  /// Volume partition iterator.
+  void forEachPartition(
```

Comment:
When we use this pattern in `osquery::Config`, we don't specify the kind of iteration. For example, there's `Config::scheduledQueries`, not `Config::mapScheduledQueries` or `Config::forEachScheduledQueries`. Minor nit would be to change this to `DeviceHelper::partitions`